### ISSUE TITLE
Rigidbodies

### ISSUE THREAD

**Issue by dovahkiinv** [CarXTechnologies/dro-map-uploader#34]
i test and rigidbody for me not work
i need mesh colider , box colider ? game market data?
to work with rigidbodies ?
i try to make hinge joint and not work

**Comment by Jeefrect** [comment]
[It works for me](https://www.youtube.com/watch?v=fIqQbtLbMbo). I hope this Unity file will help you


![image](https://github.com/user-attachments/ass
[Bumper.zip](https://github.com/user-attachments/files/17180852/Bumper.zip)
ets/01fd4b40-3803-41da-bd2b-8edc6de2445c)

**Comment by Jeefrect** [comment]
[It works for me](https://www.youtube.com/watch?v=fIqQbtLbMbo). I hope this Unity file will help you
[Bumper.zip](https://github.com/user-attachments/files/17180855/Bumper.zip)
![image](https://github.com/user-attachments/assets/b2ecdfdd-94d3-4376-a708-d1d9f388c70a)

**Comment by dovahkiinv** [comment]
> [It works for me](https://www.youtube.com/watch?v=fIqQbtLbMbo). I hope this Unity file will help you [Bumper.zip](https://github.com/user-attachments/files/17180855/Bumper.zip) ![image](https://private-user-images.githubusercontent.com/121081064/371884582-b2ecdfdd-94d3-4376-a708-d1d9f388c70a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc3MTg4NDUsIm5iZiI6MTcyNzcxODU0NSwicGF0aCI6Ii8xMjEwODEwNjQvMzcxODg0NTgyLWIyZWNkZmRkLTk0ZDMtNDM3Ni1hNzA4LWQxZDlmMzg4YzcwYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkzMFQxNzQ5MDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMWQ0ODkyZDY1NTk3ZjBmOGFiMDg2MmI0ZDEyMGFhMGFjYzdlYTgwZjE2ZGUxMmZlMmRkYTNkZmE5MjExMDAxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Sz15rMK5smpd1PjdODYkluod6IZJqRMLMFtXGHeHtkQ)

I put it on the map rigidbodies on object , in game is ok - its moved but not restart , for u this work ?

**Comment by Jeefrect** [comment]
> > [It works for me](https://www.youtube.com/watch?v=fIqQbtLbMbo). I hope this Unity file will help you [Bumper.zip](https://github.com/user-attachments/files/17180855/Bumper.zip) ![image](https://private-user-images.githubusercontent.com/121081064/371884582-b2ecdfdd-94d3-4376-a708-d1d9f388c70a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc3MTg4NDUsIm5iZiI6MTcyNzcxODU0NSwicGF0aCI6Ii8xMjEwODEwNjQvMzcxODg0NTgyLWIyZWNkZmRkLTk0ZDMtNDM3Ni1hNzA4LWQxZDlmMzg4YzcwYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkzMFQxNzQ5MDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMWQ0ODkyZDY1NTk3ZjBmOGFiMDg2MmI0ZDEyMGFhMGFjYzdlYTgwZjE2ZGUxMmZlMmRkYTNkZmE5MjExMDAxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Sz15rMK5smpd1PjdODYkluod6IZJqRMLMFtXGHeHtkQ)
> 
> I put it on the map rigidbodies on object , in game is ok - its moved but not restart , for u this work ?

Need a script that will return objects to their default positions. There is no such script now, but it should appear in the next SDK updates maybe~~

**Comment by Mankeysocks** [comment]
Hello everyone,

I'm reaching out to see if anyone can shed some light on your bumper.zip file after I tinkered with adding a reset script. I included ValidItemData in ModMapTest.cs with a new entry: ValidItem(nameof(SimpleAutoReset), 0, 100), which allowed me to build in Unity with the script attached to the bumper on the Rigidbody.

However, while the script works perfectly in the editor, it doesn't seem to function in-game. I've also added a tag just in case. Any insights or suggestions would be greatly appreciated!

[SimpleAutoReset.zip](https://github.com/user-attachments/files/17448982/SimpleAutoReset.zip)

Thank you!

**Comment by Mankeysocks** [comment]
here was my old editor debug no its not 100% but was POC
[SimpleAutoReset_debug1.zip](https://github.com/user-attachments/files/17448980/SimpleAutoReset_debug1.zip)

**Comment by Jeefrect** [comment]
> SimpleAutoReset.zip

When you create an AssetBundle in Unity (build the map), it can contain data such as textures, models, materials, sounds, prefabs, and other assets, but it cannot contain executable code. This limitation is due to both security concerns and Unity's architecture.

If an object in an AssetBundle has a script attached to it, the object will remember which script was attached, and the values of any serializable fields (for example - public fields or fields marked with [SerializedField] in private) will be saved. However, the actual script code is not included in the AssetBundle.

**For the script to work, the following conditions must be met:**
* The script must be precompiled and included in the game build. This means the script should already be part of the game's executable files that is loaded with the game.
* The script must be available when the AssetBundle is loaded. Unity will try to find the script based on its type and GUID when you load the object from the AssetBundle to the game.
If the script is not found in the loaded build, the object will load, but the script will not work, and Unity will throw an warning indicating that it cannot find the script class.

> ValidItemData in ModMapTest.cs

There's no real point in modifying scripts in the SDK, such as adding new parameters or changing existing ones in ModMapTest.cs. The reason is that when your map is loaded into the game, the game itself also runs checks on all these parameters from ModMapTest.cs. Even though your changes might work in the Unity editor, they won't function in-game unless the game has been updated to recognize and handle those changes. This is likely why your script works in the editor but not in the game—because the game isn't aware of the new parameter you've added.

In short, any changes to scripts in the SDK, especially for in-game functionality, need corresponding updates on the game's side for them to work properly.

**Comment by Mankeysocks** [comment]
thanks for rply that was my hunch was just attempting work around

**Comment by Mankeysocks** [comment]
could we do the rest with animation path ect?
